src/packs/configuration.rs

```diff
@@ -53,11 +54,16 @@ impl Configuration {
         // This takes an unused "InitializedCacheDirectory" argument, which is a sentinel
         // value returned by
         _initialized_dir: InitializedCacheDirectory,
+        file_source: FileSource,
     ) -> Box<dyn Cache + Send + Sync> {
-        if self.cache_enabled {
-            Box::new(PerFileCache {
-                cache_dir: self.cache_directory.to_owned(),
-            })
+        if let FileSource::Disk = file_source {
```

Comment:
Each handling looks reasonable to me! We're saying that we never use the cache when piping a file from stdin, which makes a lot of sense given the context of when this command would be used!